keps/sig-instrumentation/4785-resource-state-metrics/README.md

```diff
@@ -0,0 +1,1216 @@
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+-->
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+The controller will log errors but keep reconciling for when `etcd` comes back
+up. 
+
+###### What are other known failure modes?
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+It is encouraged to __observe__ any symptoms using the telemetry metrics, while
+__monitoring__ any failure using the available set of health probes.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+The controller provides a `/debug` endpoint (on `RSM_SELF_PORT`) which
+exposes all available `pprof` data to help diagnose any issue with the binary.
+Additionally, the telemetry metrics can provide more details about the runtime
+consumptions of the binary. The health probes take into factor the health of the
+respective set of components associated with that probe and will fail if any 
+such component is not healthy (`?verbose` may be appended when querying the
+health probes' endpoint to know exactly which component(s) are not healthy).
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+## Alternatives
+
+<!--
+What other approaches did you consider, and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+We considered refactoring the Kube State Metrics' Custom Resource State API, but
+that has actually been done multiple times in the past which often amounts to
+us ending up in the same position, owing to its limited scalability.
+
```

Comment:
it could be worth also mentioning https://github.com/kubernetes/kube-state-metrics/issues/1978 as an alternative